.github/scripts/get-job-id.ts

```diff
@@ -0,0 +1,44 @@
+import { retry } from './shared/utils';
+import { z } from 'zod';
+import * as core from '@actions/core';
+
+async function main() {
+  const { Octokit } = await import('octokit');
+
+  const env = z
+    .object({
+      OWNER: z.string().default('metamask'),
+      REPOSITORY: z.string().default('metamask-extension'),
+      RUN_ID: z.coerce.number(),
+      JOB_NAME: z.string(),
+      ATTEMPT_NUMBER: z.coerce.number(),
+      GITHUB_TOKEN: z.string(),
+      GITHUB_ACTIONS: z.coerce.boolean().default(false),
+    })
+    .parse(process.env);
+
+  const github = new Octokit({ auth: env.GITHUB_TOKEN });
+
+  const job = await retry(async () => {
+    const jobs = await github.paginate(
+      github.rest.actions.listJobsForWorkflowRunAttempt,
+      {
+        owner: env.OWNER,
+        repo: env.REPOSITORY,
+        run_id: env.RUN_ID,
+        attempt_number: env.ATTEMPT_NUMBER,
+        per_page: 100,
```

Comment:
is it ever possible that our job won't be found in the first page?

Reply:
Ohhhh, I wonder if that's actually why it's been failing!  There are more jobs than 100.

Reply:
ah, I didn't think it was grabbing everything here; I assumed it was a helper for iterating over paginated APIs one by one. 

In that case, since we don't want to waste "precious compute time on CI"  😝 we should probably use the `iterator` or `map` features of the API so we don't fetch more pages than we need.